project/hychiu02/README.md

```diff
@@ -0,0 +1,96 @@
+# AutoEncoder for arrhythmia detection
+
+## Basic information
+
+[GitHub](https://github.com/hychiu02/AutoEncoder-for-arrhythmia-detection)
+
+Arrhythmia dectection using ECG signal
+
+## Problem to solve
+Heart diseases are one of the significant reasons for death all over the planet. To perform arrhythmia detection, Electrocardiograms(ECGs) is used to monitor hte functioning of heart by capturing electrical activity. Because of needs of real-time inference, the model sizeand  parameters should be small, autoencoder seems to be a good choice. I want to develop a method to build, train and inference a MLP model with C++. 

```

Comment:
Having a Python wrapper to a small library is a good way to engineer the small library.  Because the library and problem size are small, it should not be used to carry out large amount of computation.  But the (software) engineering has value.

Reply:
Maybe I misunderstand the definition of large amount computations.  Initially my thought was that Deep neural network has lots of MACs on forward pass and lots of partial derivative calculation on backward pass. And runing such model on python runtime may not be as efficient as running on C++ runtime. So I implement this method by c++ code and using it on a case. If it works fine on this case, it can be used to other different cases.

Reply:
> Maybe I misunderstand the definition of large amount computations. 

For large amount of computation, we usually refer to more than several cpu-**day**.  It is not uncommon to measure by thousands of cpu-day.  Anything that can be finished in a couple of hours on a single core is not using a large amount of computation.

But the course focuses on the software engineering for the large amount of computation, not the amount of computation itself.  You do not need to find a problem that requires the large amount of computation.  It is good enough that the problem may be extended to use a large amount of computation.  The large amount of computation needs to be justified if it really happens.

> Initially my thought was that Deep neural network has lots of MACs on forward pass and lots of partial derivative calculation on backward pass. And runing such model on python runtime may not be as efficient as running on C++ runtime. 

This is correct.  But IIUC you are doing inferencing (correct me if I am wrong).  Inferential computation is much less than training.

> So I implement this method by c++ code and using it on a case. If it works fine on this case, it can be used to other different cases. Is it kind of large amount computation?

It is not, but it does not invalidate the problem.  It is fine to do it.

Reply:
Yeah, you're right. I just want to implemant a simple Machine Learning framework. As my example to predict VA, after training phase with backpropagation, when inferencing with only forward pass, its computation is much less than training. 
Thank you for advising! I'll try to think about the folllowing problem while I'm coding. I add my project to the list of project. Thanks for your kindly reminder.